tf2_ros/test/listener_unittest.cpp

```diff
@@ -31,40 +31,70 @@
 #include <gtest/gtest.h>
 #include <tf2_ros/transform_listener.h>
 
-
 void seed_rand()
 {
   //Seed random number generator with current microseond count
   srand(std::chrono::system_clock::now().time_since_epoch().count());
-};
+}
 
-void generate_rand_vectors(double scale, uint64_t runs, std::vector<double>& xvalues, std::vector<double>& yvalues, std::vector<double>&zvalues)
+void generate_rand_numbers(double & xvalue, double & yvalue, double & zvalue)
 {
   seed_rand();
-  for ( uint64_t i = 0; i < runs ; i++ )
-  {
-    xvalues[i] = 1.0 * ((double) rand() - (double)RAND_MAX /2.0) /(double)RAND_MAX;
-    yvalues[i] = 1.0 * ((double) rand() - (double)RAND_MAX /2.0) /(double)RAND_MAX;
-    zvalues[i] = 1.0 * ((double) rand() - (double)RAND_MAX /2.0) /(double)RAND_MAX;
-  }
+  xvalue = 1.0 * ((double) rand() - (double)RAND_MAX /2.0) /(double)RAND_MAX;
+  yvalue = 1.0 * ((double) rand() - (double)RAND_MAX /2.0) /(double)RAND_MAX;
+  zvalue = 1.0 * ((double) rand() - (double)RAND_MAX /2.0) /(double)RAND_MAX;
 }
 
+TEST(tf2_ros_test_listener, transform_listener)
+{
+  auto node = rclcpp::Node::make_shared("tf2_ros_test_listener_transform_listener");
 
-using namespace tf2;
+  rclcpp::Clock::SharedPtr clock = std::make_shared<rclcpp::Clock>(RCL_SYSTEM_TIME);
 
-TEST(tf2_ros_transform, transform_listener)
-{
-  tf2_ros::Buffer buffer;
-  tf2_ros::TransformListener tfl(buffer);
-  
+  tf2_ros::Buffer buffer(clock);
+  tf2_ros::TransformListener tfl(buffer, node, false);
 
-}
+  rclcpp::executors::SingleThreadedExecutor executor;
+  executor.add_node(node);
+  // Start spinning in a thread
+  std::thread spin_thread = std::thread(
+    std::bind(&rclcpp::executors::SingleThreadedExecutor::spin, &executor));
 
+  double x, y, z;
+  generate_rand_numbers(x, y, z);
```

Comment:
Honest question: is it any better to generate random numbers rather than just hardcode some values for x, y, and z?